autometa/taxonomy/ncbi.py

```diff
@@ -170,114 +157,113 @@ def lineage(self, taxid, canonical=True):
         return lineage
 
     def get_lineage_dataframe(self, taxids, fillna=True):
-        """Given an iterable of taxids generate a pandas DataFrame of their canonical
+        """
+        Given an iterable of taxids generate a pandas DataFrame of their canonical
         lineages.
 
+        Parameters
+        ----------
+        taxid : int
+            identifer for a taxon in the Taxonomy Database by NCBI
+        fillna : bool, optional
+            Whether to fill the empty cells  with 'unclassified' or not, default True
+
         Returns
         -------
         pd.DataFrame
             index = taxid
             columns = [superkingdom,phylum,class,order,family,genus,species]
 
-        NOTE: If you would like to merge the returned DataFrame with another
-        DataFrame... Let's say where you retrieved your taxids...
+        .. NOTE::
+            If you would like to merge the returned DataFrame with another
+            DataFrame. Let's say where you retrieved your taxids:
+
+        Example 
+        -------
 
-        merged_df = pd.merge(
-            your_df,
-            this_df,
-            how='left',
-            left_on=<taxid_column>,
-            right_index=True)
+        .. code-block:: python
 
+            merged_df = pd.merge(
+                your_df,
+                this_df,
+                how='left',
+                left_on=<taxid_column>,
+                right_index=True)
         """
         canonical_ranks = [r for r in reversed(NCBI.CANONICAL_RANKS)]
         canonical_ranks.remove('root')
         taxids = list(set(taxids))
-        taxids_ = {}
+        ranked_taxids = {}
         for rank in canonical_ranks:
             for taxid in taxids:
                 name = self.name(taxid, rank=rank)
-                if taxid not in taxids_:
-                    taxids_.update({taxid: {rank: name}})
+                if taxid not in ranked_taxids:
+                    ranked_taxids.update({taxid: {rank: name}})
                 else:
-                    taxids_[taxid].update({rank: name})
-        df = pd.DataFrame(taxids_).transpose()
+                    ranked_taxids[taxid].update({rank: name})
+        df = pd.DataFrame(ranked_taxids).transpose()
         df.index.name = 'taxid'
         if fillna:
             df.fillna(value='unclassified', inplace=True)
         return df
 
     def rank(self, taxid):
-        """Short summary.
+        """
+        Takes a taxid and returns its rank
```

Comment:
Even though I have added this line, I cannot understand how we will get an updated rank. The way I understand it is that if the `taxid` id present in `nodes.dmp` it will return the corresponding rank. So even if the `taxid` is dead we will still get the rank corresponding to that taxid in `nodes.dmp`